pkg/deployer/terraform/actuator.go

```diff
@@ -0,0 +1,179 @@
+// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package terraform
+
+import (
+	"context"
+	"reflect"
+	"time"
+
+	"github.com/gardener/landscaper/pkg/utils/kubernetes"
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/inject"
+
+	lsv1alpha1 "github.com/gardener/landscaper/pkg/apis/core/v1alpha1"
+	lsv1alpha1helper "github.com/gardener/landscaper/pkg/apis/core/v1alpha1/helper"
+	terraformv1alpha1 "github.com/gardener/landscaper/pkg/apis/deployer/terraform/v1alpha1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+// OperationType defines the value of a Operation that is propagated to the container.
+type OperationType string
+
+const (
+	// OperationReconcile is the value of the Operation that defines a reconcile operation.
+	OperationReconcile OperationType = "RECONCILE"
+
+	// OperationDelete is the value of the Operation that defines a delete operation.
+	OperationDelete OperationType = "DELETE"
+)
+
+func NewActuator(log logr.Logger, config *terraformv1alpha1.Configuration) (reconcile.Reconciler, error) {
+	return &actuator{
+		log:    log,
+		config: config,
+	}, nil
+}
```

Comment:
I guess these are not needed in the terraformer deployer?

Reply:
`OperationType` is used in `reconcile` to change the logic of the Terraform reconciliation. By default it will run `apply` or `destroy` with `OperationDelete`. But there is a safe guard to avoid deleting a stack that did not succeed before.